moodler/sections.py

```diff
@@ -119,3 +124,60 @@ def get_assignments_by_section(course_id, sections_names=None, assignments_names
         assignments_by_section[section_name].append(assignment)
 
     return assignments_by_section
+
+
+def get_exercises_by_topic(courseid: int) -> Dict[str, List[Dict[str, int | str]]]:
+    """
+    Retrieves the LTI exercises and assignments for a given course by topic.
+
+    Return Example:
+    {
+        "Topic 1": {"id": 1, "name": "Assignment 1", "type": "assign"},
+        "Topic 2": {"id": 2, "name": "LTI 1", "type": "lti"},
+    }
+    """
+    course_content = core_course_get_contents(courseid)
+    course_exercises = {}
+    for section in course_content:
+        # Ignore topics with no exercises
+        if not section["modules"]:
+            continue
+
+        # Save assignments or LTI exercises only
+        section_exercises = [
+            {
+                "id": module["instance"],
+                "name": module["name"],
+                "type": module["modname"],
+            }
+            for module in section["modules"]
+            if module["modname"] in ["assign", "lti"]
+        ]
+
+        if section_exercises:
+            course_exercises[section["name"]] = section_exercises
+
+    return course_exercises
+
+
+def get_exercises(courseid: int) -> List[Dict[str, int | str]]:
+    """
+    Retrieves the LTI exercises and assignments for a given course in order.
+
+    Returns an ordered list of dictionaries that contains
+    - "id": ID of exercise in the LMS
+    - "name": Name of exercise
+    - "type": Type of exercise ("assign" | "lti")
+
+    Example:
+    [
+        {"id": 1, "name": "Assignment 1", "type": "assign"},
+        {"id": 2, "name": "LTI 1", "type": "lti"},
+    ]
+    """
+    exercises_by_topic = get_exercises_by_topic(courseid)
+    return [
+        exercise
+        for topic in exercises_by_topic
+        for exercise in exercises_by_topic[topic]
+    ]
```

Comment:
Based on `exercises_by_topic[topic]`, I'm assuming `exercises_by_topic` is a dict

```suggestion
    exercises = []
    for topic, exercises_list in test.items():
         exercises.extend(exercises_list)
     return exercises
```

Reply:
Refrain from double for loop as much as possible